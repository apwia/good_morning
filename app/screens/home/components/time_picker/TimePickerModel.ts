type TimePickerModel = {
  date: Date,
  onChange: any
};

export default TimePickerModel;