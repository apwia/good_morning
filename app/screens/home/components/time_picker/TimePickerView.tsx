import React from 'react';

import DateTimePicker from '@react-native-community/datetimepicker';

import Alarm from '../../../../model/Alarm';
import useTimePickerViewModel from './TimePickerViewModel';

type Props = {
  alarm: Alarm,
  action: any,
  whenFinish: any,
};

const TimePickerView = (props: Props) => {

  const {date, onChange} = useTimePickerViewModel(props.alarm, props.action, props.whenFinish);

  return (

    <DateTimePicker
      value={date}
      mode="time"
      is24Hour={true}
      onChange={onChange}/>

  );
  
};

export default TimePickerView;