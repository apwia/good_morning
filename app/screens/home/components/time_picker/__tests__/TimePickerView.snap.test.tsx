import 'react-native';
import React from 'react';

import renderer from 'react-test-renderer';

import TimePickerView from '../TimePickerView';
import TimePickerModel from '../TimePickerModel';

jest.mock("../TimePickerViewModel", () => {
  const fnTimePickerViewModel = jest.fn();
  fnTimePickerViewModel.mockImplementation(
    () => ({
      date: new Date(123456789),
      onChange: jest.fn()
    } as TimePickerModel),
  );

  return fnTimePickerViewModel;
  
});

import useTimePickerViewModel from '../TimePickerViewModel';


it('shoud render correctly', () => {

  const props = {
    alarm: { id: 0, time: new Date(), active: false, periodicity: 'every day' },
    action: jest.fn(),
    whenFinish: jest.fn()
  };

  const {toJSON} = renderer.create(<TimePickerView {...props}/>);
  expect(toJSON()).toMatchSnapshot();
  
});