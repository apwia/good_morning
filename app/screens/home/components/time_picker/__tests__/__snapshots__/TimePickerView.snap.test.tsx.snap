// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shoud render correctly 1`] = `
<RNDateTimePicker
  date={123456789}
  displayIOS="default"
  enabled={true}
  is24Hour={true}
  mode="time"
  onChange={[Function]}
  onPickerDismiss={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
/>
`;
